app/src/main/java/com/babylon/wallet/android/data/repository/state/AccountsStateCache.kt

```diff
@@ -94,10 +99,14 @@ class AccountsStateCache @Inject constructor(
             }
             emit(accountsToReturn)
 
+            // Do not request accounts while deleting cache
+            if (deletingCache.value) return@transform
```

Comment:
shouldn't be before the `emit(accountsToReturn)`?

Reply:
No we can still emit empty data back to the subscriber since we are deleting the wallet, we just don't have to request for the rest of them.